### PR TITLE
Orders Table with Pagination

### PR DESCRIPTION
**Pull request by NateDiamond**
This code is tested with mock data. It should be tried with a real order to see how it handles. UI could be more responsive, for example changing color based on the order's status.
Screen Shot 2023-10-23 at 5 04 52 PM

 
Screen Shot 2023-10-23 at 5 05 04 PM

### CODE REVIEW

**Comment by NateDiamond** [comment]
> **The latest updates on your projects**. Learn more about [Vercel for Git ↗︎](https://vercel.link/github-learn-more)
> 
> Name	Status	Preview	Updated (UTC)
> **btc-toolkit**	❌ Failed ([Inspect](https://vercel.com/wild-tangz/btc-toolkit/HcxwhW8naZG1LGvJA7bSSY4pFF8E))		Oct 23, 2023 9:18pm

`npm run build` passed locally for me. I'm not sure what this is.

File: src/app/api/orders/route.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { NextResponse } from 'next/server';
+
+import prisma from '../../../prisma/prisma.mjs';
+
+const DEFAULT_TAKE = 10;
+
+export async function GET({nextUrl: {searchParams}}) {
+  var response = {}
+  try {
+    if(searchParams.has('getCount') && searchParams.get('getCount') == 'true') {
```
Comment (thaddeusdiamond) [inline src/app/api/orders/route.js]: minor: space after `if`

File: src/app/api/orders/route.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      response['count'] = await prisma.order.count();
+    }
+    const cursor = searchParams.has('cursor') ? searchParams.get('cursor') : 'null';
+    const take = searchParams.has('take') ? parseInt(searchParams.get('take')) : DEFAULT_TAKE;
+    var orders;
+    if (cursor != 'null') {
+      console.log("here");
+      orders = await prisma.order.findMany({
```
Comment (thaddeusdiamond) [inline src/app/api/orders/route.js:21]: Great.  This is what I was thinking... but do you need a `where` clause to specify the address? Should be a parameter to this backend.

File: src/components/orderstable/orderstable.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
+import {useState, useEffect} from 'react';
+
+const NUM_ROWS = 3;
+
+export function OrdersTable() {
+  const [pages, setPages] = useState({});
+  const [currentPage, setCurrentPage] = useState(0); // 0-indexed
+  const [orders, setOrders] = useState(null);
+  const [ordersCount, setOrdersCount] = useState(0)
+
```
Comment (thaddeusdiamond) [inline src/components/orderstable/orderstable.js:10]: extra newline

File: src/components/orderstable/orderstable.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    setOrdersCount(responseJSON['count']);
+    updatePages(currentPage, initialPage);
+    console.log(pages);
+    setOrders(initialPage);
+  }
+
+  useEffect(initialLoadPage, []);
+  
+  async function loadNextPage() {
+    if((currentPage + 1)*NUM_ROWS >= ordersCount) {
```
Comment (thaddeusdiamond) [inline src/components/orderstable/orderstable.js]: spacing is odd here

File: src/components/orderstable/orderstable.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      return;
+    }
+    const includeCount = false
+    var cursor, nextPage;
+    cursor = orders.slice(-1)[0]['id'];
+    if(`${currentPage+1}` in pages) {
```
Comment (thaddeusdiamond) [inline src/components/orderstable/orderstable.js]: space after if

File: src/components/orderstable/orderstable.js [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      const responseJSON = await apiResponse.json();
+      nextPage = responseJSON['orders'];
+      updatePages(currentPage+1, nextPage);
+    }
+    setCurrentPage(currentPage+1);
+    setOrders(nextPage);
+  }
+
+  function loadPreviousPage() {
+    if(currentPage == 0)  {
```
Comment (thaddeusdiamond) [inline src/components/orderstable/orderstable.js]: space after if

File: src/components/orderstable/orderstable.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      </div>
+      <div>
+        <nav class="isolate inline-flex -space-x-px rounded-md shadow-sm" aria-label="Pagination">
+          <a onClick={loadPreviousPage} class="relative inline-flex items-center rounded-l-md px-2 py-2 text-gray-300 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker focus:z-20 focus:outline-offset-0">
+            <span class="sr-only">Previous</span>
+            <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+              <path fill-rule="evenodd" d="M12.79 5.23a.75.75 0 01-.02 1.06L8.832 10l3.938 3.71a.75.75 0 11-1.04 1.08l-4.5-4.25a.75.75 0 010-1.08l4.5-4.25a.75.75 0 011.06.02z" clip-rule="evenodd" />
+            </svg>
+          </a>
+          {/* <!-- Current: "z-10 bg-indigo-600 text-white focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600", Default: "text-gray-900 ring-1 ring-inset ring-gray-300 hover:bg-gray-50 focus:outline-offset-0" --> */}
```
Comment (thaddeusdiamond) [inline src/components/orderstable/orderstable.js]: Delete commented code

File: src/components/orderstable/orderstable.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
+import {useState, useEffect} from 'react';
+
+const NUM_ROWS = 3;
+
+export function OrdersTable() {
```
Comment (thaddeusdiamond) [inline src/components/orderstable/orderstable.js]: You should add a parameter here for `address` then send it to the backend as a data source to populate.  The `address` can be pulled from `orderData.get("walletAddr")` (see page.js for sample codes).

That object is automatically managed/updated using react's `useState`

**Review by thaddeusdiamond — COMMENTED** [review]
Exactly the right path to go down.  Keep trucking

File: src/components/orderstable/orderstable.js (hunk 8 above)
Comment (NateDiamond) [reply src/components/orderstable/orderstable.js]: Okay added, but I would like to figure out a way to test.

**Comment by thaddeusdiamond** [comment]
Check out this patch... that seemed to work for me in showing up correctly (also includes some changes with dark mode you should take a look at):

```
diff --git a/src/app/page.js b/src/app/page.js
index 6b7b7da..85bb9f5 100644
--- a/src/app/page.js
+++ b/src/app/page.js
@@ -238,17 +238,17 @@ export default function Home() {
       </div>
       <div class="mx-auto max-w-3xl px-4 py-5 sm:px-6 lg:max-w-7xl lg:px-8">
         <div class="mx-auto max-w-7xl">
-          <div class="bg-gray-700 py-10 rounded-lg">
+          <div class="rounded-lg shadow bg-white dark:bg-gray-700 py-8 rounded-lg">
             <div class="px-4 sm:px-6 lg:px-8">
               <div class="sm:flex sm:items-center">
                 <div class="sm:flex-auto">
-                  <h1 class="text-base font-semibold leading-6 text-white text-center">My Orders</h1>
+                  <h1 class="text-tangz-blue font-semibold leading-6 dark:text-white text-center">My Orders</h1>
                 </div>
               </div>
-              <div class="mt-8 flow-root">
+              <div class="mt-4 flow-root">
                 <div class="-mx-4 -my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
                   <div class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8">
-                    <OrdersTable address={orderData.get('walletAddr')}/>
+                    <OrdersTable orderData={orderData} />
                   </div>
                 </div>
               </div>
diff --git a/src/components/orderstable/orderstable.js b/src/components/orderstable/orderstable.js
index e10e94b..8257333 100644
--- a/src/components/orderstable/orderstable.js
+++ b/src/components/orderstable/orderstable.js
@@ -2,34 +2,32 @@ import {useState, useEffect} from 'react';
 
 const NUM_ROWS = 3;
 
-export function OrdersTable(address) {
+export function OrdersTable({ orderData }) {
   const [pages, setPages] = useState({});
   const [currentPage, setCurrentPage] = useState(0); // 0-indexed
   const [orders, setOrders] = useState(null);
   const [ordersCount, setOrdersCount] = useState(0);
-  
+
   function updatePages(pageNumber, pageContent) {
-    var currPages = pages;
-    currPages[`${pageNumber}`] = pageContent;
-    setPages(currPages);
+    pages[`${pageNumber}`] = pageContent;
+    setPages({...pages});
   }
 
-  async function initialLoadPage() {
-    var cursor, includeCount, initialPage;
-    cursor = null;
-    includeCount = true;
-    const apiCall = `/api/orders?address=${address}&cursor=${cursor}&take=${NUM_ROWS}&getCount=${includeCount}`
-    const apiResponse = await fetch(apiCall);
-    const responseJSON = await apiResponse.json();
-    initialPage = responseJSON['orders'];
-    setOrdersCount(responseJSON['count']);
-    updatePages(currentPage, initialPage);
-    console.log(pages);
-    setOrders(initialPage);
+  function initialLoadPage(address) {
+    fetch(`/api/orders?address=${address}&cursor=null&take=${NUM_ROWS}&getCount=true`)
+      .then(resp => resp.json())
+      .then(responseJSON => {
+        const initialPage = responseJSON['orders'];
+        setOrdersCount(responseJSON['count']);
+        updatePages(currentPage, initialPage);
+        setOrders(initialPage);
+      });
   }
 
-  useEffect(initialLoadPage, []);
-  
+  useEffect(() => {
+    initialLoadPage(orderData.get("walletAddr"))
+  }, [orderData]);
+
   async function loadNextPage() {
     if ((currentPage + 1) * NUM_ROWS >= ordersCount) {
       return;
@@ -59,70 +57,68 @@ export function OrdersTable(address) {
   }
 
   return (
-  <div>
-    <table class="min-w-full divide-y gray-300">
-      <thead>
-        <tr class="bg-tangz-blue-darker">
-          <th scope="col" class="px-3 py-3.5 pl-4 pr-3 text-left text-sm font-semibold text-white">Order ID</th>
-          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Price</th>
-          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Service Fee</th>
-          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Status</th>
-          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Created At</th>
-          <th scope="col" class="relative py-3.5 pl-3 pr-4 sm:pr-0">
-            <span class="sr-only">Edit</span>
-          </th>
-        </tr>
-      </thead>
-      <tbody class="divide-y divide-gray-800">
-        {orders != null ? orders.map( (order) => (
-          <tr class="bg-gray-700" key={order}>
-            <td class="whitespace-nowrap px-3 py-4 pl-4 pr-3 text-sm font-medium text-white">{order['id']}</td>
-            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['price']}</td>
-            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['service_fee']}</td>
-            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['status']}</td>
-            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['createdAt']}</td>
-            <td class="relative whitespace-nowrap py-4 pl-3 pr-4 text-right text-sm font-medium sm:pr-0">
-            </td>
+    <div>
+      <table class="min-w-full divide-y gray-300">
+        <thead>
+          <tr class="bg-tangz-blue-darker">
+            <th scope="col" class="px-3 py-3.5 pl-4 pr-3 text-left text-sm font-semibold text-white">Order ID</th>
+            <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Cost (sats)</th>
+            <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Status</th>
+            <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Created At</th>
+            <th scope="col" class="relative py-3.5 pl-3 pr-4 sm:pr-0">
+              <span class="sr-only">Edit</span>
+            </th>
           </tr>
-        )): ""}
-      </tbody>
-    </table>
-    <div class="flex items-center justify-between border-t border-gray-200 bg-gray-700 px-4 py-3 sm:px-6">
-    <div class="flex flex-1 justify-between sm:hidden">
-      <a href="#" class="relative inline-flex items-center rounded-md border border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Previous</a>
-      <a href="#" class="relative ml-3 inline-flex items-center rounded-md border border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Next</a>
-    </div>
-    <div class="hidden sm:flex sm:flex-1 sm:items-center sm:justify-between">
-      <div>
-        <p class="text-sm text-gray-300">
-          Showing
-          <span class="font-medium"> {ordersCount ? (currentPage * NUM_ROWS) + 1 : 0} </span>
-          to
-          <span class="font-medium"> {Math.min((currentPage + 1) * NUM_ROWS, ordersCount)} </span>
-          of
-          <span class="font-medium"> {ordersCount} </span>
-          results
-        </p>
-      </div>
-      <div>
-        <nav class="isolate inline-flex -space-x-px rounded-md shadow-sm" aria-label="Pagination">
-          <a onClick={loadPreviousPage} class="relative inline-flex items-center rounded-l-md px-2 py-2 text-gray-300 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker focus:z-20 focus:outline-offset-0">
-            <span class="sr-only">Previous</span>
-            <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
-              <path fill-rule="evenodd" d="M12.79 5.23a.75.75 0 01-.02 1.06L8.832 10l3.938 3.71a.75.75 0 11-1.04 1.08l-4.5-4.25a.75.75 0 010-1.08l4.5-4.25a.75.75 0 011.06.02z" clip-rule="evenodd" />
-            </svg>
-          </a>
-          <a class="relative inline-flex items-center px-4 py-2 text-sm font-semibold text-gray-300 ring-1 ring-inset ring-gray-300 focus:z-20 focus:outline-offset-0">{currentPage+1}</a>
-          <a onClick={loadNextPage} class="relative inline-flex items-center rounded-r-md px-2 py-2 text-gray-300 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker focus:z-20 focus:outline-offset-0">
-            <span class="sr-only">Next</span>
-            <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
-              <path fill-rule="evenodd" d="M7.21 14.77a.75.75 0 01.02-1.06L11.168 10 7.23 6.29a.75.75 0 111.04-1.08l4.5 4.25a.75.75 0 010 1.08l-4.5 4.25a.75.75 0 01-1.06-.02z" clip-rule="evenodd" />
-            </svg>
-          </a>
-        </nav>
+        </thead>
+        <tbody class="divide-y divide-gray-700 dark:divide-gray-300">
+          {orders != null ? orders.map( (order) => (
+            <tr key={order}>
+              <td class="whitespace-nowrap px-3 py-4 pl-4 pr-3 text-sm font-medium text-gray-700 dark:text-white">{order['id']}</td>
+              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-700 dark:text-white">{order['price']}</td>
+              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-700 dark:text-white">{order['status']}</td>
+              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-700 dark:text-white">{order['createdAt']}</td>
+              <td class="relative whitespace-nowrap py-4 pl-3 pr-4 text-right text-sm font-medium sm:pr-0">
+              </td>
+            </tr>
+          )): ""}
+        </tbody>
+      </table>
+      <div class="flex items-center justify-between border-t border-gray-700 dark:border-gray-300 px-4 py-3 sm:px-6">
+        <div class="flex flex-1 justify-between sm:hidden">
+          <a href="#" class="relative inline-flex items-center rounded-md border dark:border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Previous</a>
+          <a href="#" class="relative ml-3 inline-flex items-center rounded-md border dark:border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Next</a>
+        </div>
+        <div class="hidden sm:flex sm:flex-1 sm:items-center sm:justify-between">
+          <div>
+            <p class="text-sm text-gray-700 dark:text-white">
+              Showing
+              <span class="font-medium"> {ordersCount ? (currentPage * NUM_ROWS) + 1 : 0} </span>
+              to
+              <span class="font-medium"> {Math.min((currentPage + 1) * NUM_ROWS, ordersCount)} </span>
+              of
+              <span class="font-medium"> {ordersCount} </span>
+              results
+            </p>
+          </div>
+          <div>
+            <nav class="isolate inline-flex -space-x-px rounded-md shadow-sm text-gray-700 dark:text-white" aria-label="Pagination">
+              <a onClick={loadPreviousPage} class="relative inline-flex items-center rounded-l-md px-2 py-2 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker hover:text-white focus:z-20 focus:outline-offset-0">
+                <span class="sr-only">Previous</span>
+                <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+                  <path fill-rule="evenodd" d="M12.79 5.23a.75.75 0 01-.02 1.06L8.832 10l3.938 3.71a.75.75 0 11-1.04 1.08l-4.5-4.25a.75.75 0 010-1.08l4.5-4.25a.75.75 0 011.06.02z" clip-rule="evenodd" />
+                </svg>
+              </a>
+              <a class="relative inline-flex items-center px-4 py-2 text-sm font-semibold ring-1 ring-inset ring-gray-300 focus:z-20 focus:outline-offset-0">{currentPage+1}</a>
+              <a onClick={loadNextPage} class="relative inline-flex items-center rounded-r-md px-2 py-2 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker hover:text-white focus:z-20 focus:outline-offset-0">
+                <span class="sr-only">Next</span>
+                <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+                  <path fill-rule="evenodd" d="M7.21 14.77a.75.75 0 01.02-1.06L11.168 10 7.23 6.29a.75.75 0 111.04-1.08l4.5 4.25a.75.75 0 010 1.08l-4.5 4.25a.75.75 0 01-1.06-.02z" clip-rule="evenodd" />
+                </svg>
+              </a>
+            </nav>
+          </div>
+        </div>
       </div>
     </div>
-  </div>
-  </div>
   );
-}
\ No newline at end of file
+}
```

**Comment by NateDiamond** [comment]
> Check out this patch... that seemed to work for me in showing up correctly (also includes some changes with dark mode you should take a look at):
> 
> ```
> diff --git a/src/app/page.js b/src/app/page.js
> index 6b7b7da..85bb9f5 100644
> --- a/src/app/page.js
> +++ b/src/app/page.js
> @@ -238,17 +238,17 @@ export default function Home() {
>        </div>
>        <div class="mx-auto max-w-3xl px-4 py-5 sm:px-6 lg:max-w-7xl lg:px-8">
>          <div class="mx-auto max-w-7xl">
> -          <div class="bg-gray-700 py-10 rounded-lg">
> +          <div class="rounded-lg shadow bg-white dark:bg-gray-700 py-8 rounded-lg">
>              <div class="px-4 sm:px-6 lg:px-8">
>                <div class="sm:flex sm:items-center">
>                  <div class="sm:flex-auto">
> -                  <h1 class="text-base font-semibold leading-6 text-white text-center">My Orders</h1>
> +                  <h1 class="text-tangz-blue font-semibold leading-6 dark:text-white text-center">My Orders</h1>
>                  </div>
>                </div>
> -              <div class="mt-8 flow-root">
> +              <div class="mt-4 flow-root">
>                  <div class="-mx-4 -my-2 overflow-x-auto sm:-mx-6 lg:-mx-8">
>                    <div class="inline-block min-w-full py-2 align-middle sm:px-6 lg:px-8">
> -                    <OrdersTable address={orderData.get('walletAddr')}/>
> +                    <OrdersTable orderData={orderData} />
>                    </div>
>                  </div>
>                </div>
> diff --git a/src/components/orderstable/orderstable.js b/src/components/orderstable/orderstable.js
> index e10e94b..8257333 100644
> --- a/src/components/orderstable/orderstable.js
> +++ b/src/components/orderstable/orderstable.js
> @@ -2,34 +2,32 @@ import {useState, useEffect} from 'react';
>  
>  const NUM_ROWS = 3;
>  
> -export function OrdersTable(address) {
> +export function OrdersTable({ orderData }) {
>    const [pages, setPages] = useState({});
>    const [currentPage, setCurrentPage] = useState(0); // 0-indexed
>    const [orders, setOrders] = useState(null);
>    const [ordersCount, setOrdersCount] = useState(0);
> -  
> +
>    function updatePages(pageNumber, pageContent) {
> -    var currPages = pages;
> -    currPages[`${pageNumber}`] = pageContent;
> -    setPages(currPages);
> +    pages[`${pageNumber}`] = pageContent;
> +    setPages({...pages});
>    }
>  
> -  async function initialLoadPage() {
> -    var cursor, includeCount, initialPage;
> -    cursor = null;
> -    includeCount = true;
> -    const apiCall = `/api/orders?address=${address}&cursor=${cursor}&take=${NUM_ROWS}&getCount=${includeCount}`
> -    const apiResponse = await fetch(apiCall);
> -    const responseJSON = await apiResponse.json();
> -    initialPage = responseJSON['orders'];
> -    setOrdersCount(responseJSON['count']);
> -    updatePages(currentPage, initialPage);
> -    console.log(pages);
> -    setOrders(initialPage);
> +  function initialLoadPage(address) {
> +    fetch(`/api/orders?address=${address}&cursor=null&take=${NUM_ROWS}&getCount=true`)
> +      .then(resp => resp.json())
> +      .then(responseJSON => {
> +        const initialPage = responseJSON['orders'];
> +        setOrdersCount(responseJSON['count']);
> +        updatePages(currentPage, initialPage);
> +        setOrders(initialPage);
> +      });
>    }
>  
> -  useEffect(initialLoadPage, []);
> -  
> +  useEffect(() => {
> +    initialLoadPage(orderData.get("walletAddr"))
> +  }, [orderData]);
> +
>    async function loadNextPage() {
>      if ((currentPage + 1) * NUM_ROWS >= ordersCount) {
>        return;
> @@ -59,70 +57,68 @@ export function OrdersTable(address) {
>    }
>  
>    return (
> -  <div>
> -    <table class="min-w-full divide-y gray-300">
> -      <thead>
> -        <tr class="bg-tangz-blue-darker">
> -          <th scope="col" class="px-3 py-3.5 pl-4 pr-3 text-left text-sm font-semibold text-white">Order ID</th>
> -          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Price</th>
> -          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Service Fee</th>
> -          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Status</th>
> -          <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Created At</th>
> -          <th scope="col" class="relative py-3.5 pl-3 pr-4 sm:pr-0">
> -            <span class="sr-only">Edit</span>
> -          </th>
> -        </tr>
> -      </thead>
> -      <tbody class="divide-y divide-gray-800">
> -        {orders != null ? orders.map( (order) => (
> -          <tr class="bg-gray-700" key={order}>
> -            <td class="whitespace-nowrap px-3 py-4 pl-4 pr-3 text-sm font-medium text-white">{order['id']}</td>
> -            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['price']}</td>
> -            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['service_fee']}</td>
> -            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['status']}</td>
> -            <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-300">{order['createdAt']}</td>
> -            <td class="relative whitespace-nowrap py-4 pl-3 pr-4 text-right text-sm font-medium sm:pr-0">
> -            </td>
> +    <div>
> +      <table class="min-w-full divide-y gray-300">
> +        <thead>
> +          <tr class="bg-tangz-blue-darker">
> +            <th scope="col" class="px-3 py-3.5 pl-4 pr-3 text-left text-sm font-semibold text-white">Order ID</th>
> +            <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Cost (sats)</th>
> +            <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Status</th>
> +            <th scope="col" class="px-3 py-3.5 text-left text-sm font-semibold text-white">Created At</th>
> +            <th scope="col" class="relative py-3.5 pl-3 pr-4 sm:pr-0">
> +              <span class="sr-only">Edit</span>
> +            </th>
>            </tr>
> -        )): ""}
> -      </tbody>
> -    </table>
> -    <div class="flex items-center justify-between border-t border-gray-200 bg-gray-700 px-4 py-3 sm:px-6">
> -    <div class="flex flex-1 justify-between sm:hidden">
> -      <a href="#" class="relative inline-flex items-center rounded-md border border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Previous</a>
> -      <a href="#" class="relative ml-3 inline-flex items-center rounded-md border border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Next</a>
> -    </div>
> -    <div class="hidden sm:flex sm:flex-1 sm:items-center sm:justify-between">
> -      <div>
> -        <p class="text-sm text-gray-300">
> -          Showing
> -          <span class="font-medium"> {ordersCount ? (currentPage * NUM_ROWS) + 1 : 0} </span>
> -          to
> -          <span class="font-medium"> {Math.min((currentPage + 1) * NUM_ROWS, ordersCount)} </span>
> -          of
> -          <span class="font-medium"> {ordersCount} </span>
> -          results
> -        </p>
> -      </div>
> -      <div>
> -        <nav class="isolate inline-flex -space-x-px rounded-md shadow-sm" aria-label="Pagination">
> -          <a onClick={loadPreviousPage} class="relative inline-flex items-center rounded-l-md px-2 py-2 text-gray-300 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker focus:z-20 focus:outline-offset-0">
> -            <span class="sr-only">Previous</span>
> -            <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
> -              <path fill-rule="evenodd" d="M12.79 5.23a.75.75 0 01-.02 1.06L8.832 10l3.938 3.71a.75.75 0 11-1.04 1.08l-4.5-4.25a.75.75 0 010-1.08l4.5-4.25a.75.75 0 011.06.02z" clip-rule="evenodd" />
> -            </svg>
> -          </a>
> -          <a class="relative inline-flex items-center px-4 py-2 text-sm font-semibold text-gray-300 ring-1 ring-inset ring-gray-300 focus:z-20 focus:outline-offset-0">{currentPage+1}</a>
> -          <a onClick={loadNextPage} class="relative inline-flex items-center rounded-r-md px-2 py-2 text-gray-300 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker focus:z-20 focus:outline-offset-0">
> -            <span class="sr-only">Next</span>
> -            <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
> -              <path fill-rule="evenodd" d="M7.21 14.77a.75.75 0 01.02-1.06L11.168 10 7.23 6.29a.75.75 0 111.04-1.08l4.5 4.25a.75.75 0 010 1.08l-4.5 4.25a.75.75 0 01-1.06-.02z" clip-rule="evenodd" />
> -            </svg>
> -          </a>
> -        </nav>
> +        </thead>
> +        <tbody class="divide-y divide-gray-700 dark:divide-gray-300">
> +          {orders != null ? orders.map( (order) => (
> +            <tr key={order}>
> +              <td class="whitespace-nowrap px-3 py-4 pl-4 pr-3 text-sm font-medium text-gray-700 dark:text-white">{order['id']}</td>
> +              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-700 dark:text-white">{order['price']}</td>
> +              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-700 dark:text-white">{order['status']}</td>
> +              <td class="whitespace-nowrap px-3 py-4 text-sm text-gray-700 dark:text-white">{order['createdAt']}</td>
> +              <td class="relative whitespace-nowrap py-4 pl-3 pr-4 text-right text-sm font-medium sm:pr-0">
> +              </td>
> +            </tr>
> +          )): ""}
> +        </tbody>
> +      </table>
> +      <div class="flex items-center justify-between border-t border-gray-700 dark:border-gray-300 px-4 py-3 sm:px-6">
> +        <div class="flex flex-1 justify-between sm:hidden">
> +          <a href="#" class="relative inline-flex items-center rounded-md border dark:border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Previous</a>
> +          <a href="#" class="relative ml-3 inline-flex items-center rounded-md border dark:border-gray-300 bg-gray-700 px-4 py-2 text-sm font-medium text-gray-300 hover:bg-gray-50">Next</a>
> +        </div>
> +        <div class="hidden sm:flex sm:flex-1 sm:items-center sm:justify-between">
> +          <div>
> +            <p class="text-sm text-gray-700 dark:text-white">
> +              Showing
> +              <span class="font-medium"> {ordersCount ? (currentPage * NUM_ROWS) + 1 : 0} </span>
> +              to
> +              <span class="font-medium"> {Math.min((currentPage + 1) * NUM_ROWS, ordersCount)} </span>
> +              of
> +              <span class="font-medium"> {ordersCount} </span>
> +              results
> +            </p>
> +          </div>
> +          <div>
> +            <nav class="isolate inline-flex -space-x-px rounded-md shadow-sm text-gray-700 dark:text-white" aria-label="Pagination">
> +              <a onClick={loadPreviousPage} class="relative inline-flex items-center rounded-l-md px-2 py-2 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker hover:text-white focus:z-20 focus:outline-offset-0">
> +                <span class="sr-only">Previous</span>
> +                <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
> +                  <path fill-rule="evenodd" d="M12.79 5.23a.75.75 0 01-.02 1.06L8.832 10l3.938 3.71a.75.75 0 11-1.04 1.08l-4.5-4.25a.75.75 0 010-1.08l4.5-4.25a.75.75 0 011.06.02z" clip-rule="evenodd" />
> +                </svg>
> +              </a>
> +              <a class="relative inline-flex items-center px-4 py-2 text-sm font-semibold ring-1 ring-inset ring-gray-300 focus:z-20 focus:outline-offset-0">{currentPage+1}</a>
> +              <a onClick={loadNextPage} class="relative inline-flex items-center rounded-r-md px-2 py-2 ring-1 ring-inset ring-gray-300 hover:bg-tangz-blue-darker hover:text-white focus:z-20 focus:outline-offset-0">
> +                <span class="sr-only">Next</span>
> +                <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
> +                  <path fill-rule="evenodd" d="M7.21 14.77a.75.75 0 01.02-1.06L11.168 10 7.23 6.29a.75.75 0 111.04-1.08l4.5 4.25a.75.75 0 010 1.08l-4.5 4.25a.75.75 0 01-1.06-.02z" clip-rule="evenodd" />
> +                </svg>
> +              </a>
> +            </nav>
> +          </div>
> +        </div>
>        </div>
>      </div>
> -  </div>
> -  </div>
>    );
> -}
> \ No newline at end of file
> +}
> ```

LGTM! I might have leftover a `span` tag that says `Edit`, just FYI.

**Review by thaddeusdiamond — APPROVED** [review]
Way to go.  Merging and closing.

**Pull request merged**